src/sql/workbench/api/common/vscodeExecuteProvider.ts

```diff
@@ -0,0 +1,313 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type * as vscode from 'vscode';
+import type * as azdata from 'azdata';
+import { ADSNotebookController } from 'sql/workbench/api/common/adsNotebookController';
+import * as nls from 'vs/nls';
+
+class VSCodeFuture implements azdata.nb.IFuture {
+	private _inProgress = true;
+
+	constructor(private readonly _executeCompletion: Promise<void>) {
+	}
+
+	dispose() {
+		// No-op
+	}
+
+	public get inProgress(): boolean {
+		return this._inProgress;
+	}
+
+	public set inProgress(value: boolean) {
+		this._inProgress = value;
+	}
+
+	public get msg(): azdata.nb.IMessage | undefined {
+		return undefined;
+	}
+
+	public get done(): Thenable<azdata.nb.IShellMessage | undefined> {
+		return this._executeCompletion.then(() => {
+			return undefined;
+		}).finally(() => {
+			this._inProgress = false;
+		});
+	}
+
+	setReplyHandler(handler: azdata.nb.MessageHandler<azdata.nb.IShellMessage>): void {
+		// No-op
+	}
+
+	setStdInHandler(handler: azdata.nb.MessageHandler<azdata.nb.IStdinMessage>): void {
+		// No-op
+	}
+
+	setIOPubHandler(handler: azdata.nb.MessageHandler<azdata.nb.IIOPubMessage>): void {
+		// No-op
+	}
+
+	registerMessageHook(hook: (msg: azdata.nb.IIOPubMessage) => boolean | Thenable<boolean>): void {
+		// No-op
+	}
+
+	removeMessageHook(hook: (msg: azdata.nb.IIOPubMessage) => boolean | Thenable<boolean>): void {
+		// No-op
+	}
+
+	sendInputReply(content: azdata.nb.IInputReply): void {
+		// No-op
+	}
+}
+
+class VSCodeKernel implements azdata.nb.IKernel {
```

Comment:
It would be nice to add a short doc comment for these classes explaining what they're used for.